### PR TITLE
Add percent change load action

### PR DESCRIPTION
**Pull request by bc-rte**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

https://github.com/powsybl/powsybl-core/issues/3078

**What kind of change does this PR introduce?**

This PR adds a new action that allows users to change a load without knowing existing values, by specifying a percent P0 change and its impact on Q0.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

### CODE REVIEW

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/PctLoadModification.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @Override
+    public String getName() {
+        return "pctLoadModification";
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        Load load = network.getLoad(loadId);
+        if (load == null) {
+            throw new PowsyblException(getMessageHeader() + ": Tried to apply modification on " + loadId + " but no load was found");
```
[the remark this answers is not in the document]
Comment (bc-rte) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/PctLoadModification.java:50]: Wouldn't you find that the suggested message is not as precise?

If so, wouldn't it be better to favor that over consistency?

File: iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/PctLoadModificationTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        Network network = EurostagTutorialExample1Factory.create();
+        Load load = network.getLoad("LOAD");
+        assertEquals(200.0, load.getQ0());
+        PctLoadModification modification = new PctLoadModification("LOAD", 0, 2.5);
+        modification.apply(network);
+        assertEquals(205.0, load.getQ0());
+    }
+
+    @Test
+    void shouldThrowWhenLoadDecreasesTooMuch() {
```
[the remark this answers is not in the document]
Comment (bc-rte) [reply iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/PctLoadModificationTest.java:59]: First case added.

Not sure what behaviour I should test in the second one.

[the remark this answers is not in the document]
Comment (olperr1) [reply iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/PctLoadModificationTest.java]: Sorry, it wasn't clear enough. The second test I thought of was to check that when both arguments have non-zero values, both changes are applied.

**Comment by olperr1** [comment]
For the documentation, I suggest the following change in [`docs/simulation/security/index.md`](https://github.com/powsybl/powsybl-core/blob/fa2b1434bc56ab9bb4ecd5fb932339f4b61a6bc1/docs/simulation/security/index.md?plain=1#L42):

```diff
- - Change the active and/or reactive power of a load
+ - Change the active and/or reactive power of a load (by setting the values, or defining changes in value or percentage)
```

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/PercentChangeLoadModification.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public String getName() {
+        return "pctLoadModification";
+    }
+
+    @Override
+    public void apply(Network network, NamingStrategy namingStrategy, boolean throwException, ComputationManager computationManager, ReportNode reportNode) {
+        Load load = network.getLoad(loadId);
+        if (load == null) {
+            throw new PowsyblException(getMessageHeader() + ": Tried to apply modification on " + loadId + " but no load was found");
+        }
```
Comment (olperr1) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/PercentChangeLoadModification.java:52]: In fact, the exception should only be thrown when `throwException` is `true`.

There's a method for this:
```suggestion
        if (load == null) {
            logOrThrow(throwException, "Load '" + loadId + "' not found");
            return;
        }
```

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/PercentChangeLoadModificationTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+class PercentChangeLoadModificationTest {
+
+    @Test
+    void shouldIncreaseP0() {
+        Network network = EurostagTutorialExample1Factory.create();
+        Load load = network.getLoad("LOAD");
+        assertEquals(600.0, load.getP0());
+        PercentChangeLoadModification modification = new PercentChangeLoadModification("LOAD", 3.51, 0);
+        modification.apply(network);
+        assertEquals(621.06, load.getP0());
```
Comment (olperr1) [inline iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/PercentChangeLoadModificationTest.java:32]: When comparing doubles, it is a good practice to use a delta:

```suggestion
        assertEquals(621.06, load.getP0(), 0.01);
```

**Review by olperr1 — CHANGES_REQUESTED** [review]

**Review by alicecaron — APPROVED** [review]

**Pull request merged**